webserver/views/api/v1/test/test_datasets.py

```diff
@@ -79,3 +79,648 @@ def test_create_dataset(self, create_from_dict):
         self.assertEqual(resp.status_code, 200)
         expected = {"success": True, "dataset_id": "6b6b9205-f9c8-4674-92f5-2ae17bcb3cb0"}
         self.assertEqual(resp.json, expected)
+
+
+    def test_add_recordings(self):
```

Comment:
With tests in the webserver, we don't need to create data in the database and check that it has been updated. Instead, we can use mocks, which verify that we call a method with the correct arguments, but doesn't actually call the function. See some examples here: https://github.com/metabrainz/acousticbrainz-server/blob/master/webserver/views/api/v1/test/test_datasets.py#L48-L63

With mocks we can throw exceptions (using `.side_effect`) to make sure that the API returns these errors correctly, and we can also return data (`.return_value`) if we want to pass some data back to the API.

You can read more documentation on mock here: https://docs.python.org/dev/library/unittest.mock.html (don't worry that it says for python 3.3, we use a version of the same library for python 2)